### ISSUE TITLE
SecureStream is hard coded into StreamEncryption

### ISSUE THREAD

**Issue by DangerLifter** [reactphp/socket-client#30]
SecureStream is hard coded into StreamEncryption and there is no easy ability to replace it with extended class.
Reason for replacement: I need to catch and correctly proceed exception " stream_get_contents(): SSL: Connection reset by peer " in method SecureStream::handleData(); 
Or would be nice updte code with correct catch;

```
        try {
            $this->handleData($stream);
        } catch (\Exception $e) {
            $this->emit('error', array($e, $this));
        }
```

\React\Stream\Stream

**Comment by cboden** [comment]
Could you provide some code that triggers this error you're having? We're currently looking into issues with `stream_get_contents` but are having difficulty reproducing errors.

**Comment by clue** [comment]
This issue will likely also be addressed via https://github.com/reactphp/stream/pull/42 and https://github.com/reactphp/stream/pull/40.

**Comment by clue** [comment]
Closed via #72
